app/grants/tasks.py

```diff
@@ -118,78 +120,202 @@ def process_grant_contribution(self, grant_id, grant_slug, profile_id, package,
     :return:
     """
     from grants.views import record_subscription_activity_helper
-
-    grant = Grant.objects.get(pk=grant_id)
     profile = Profile.objects.get(pk=profile_id)
+    subscriptions = []
+    successes = []
+    failures = []
+    for (index, grant_id) in enumerate(grant_ids_list):
+        try:
+            grant = Grant.objects.get(pk=grant_id)
+        except Grant.DoesNotExist:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant Does Not Exist'),
+                'grant':grant_id,
+                'text': _('This grant does not exist'),
+                'subtext': _('No grant with this ID was found'),
+                'success': False
+            })
+            continue
+
+        if not grant.active:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant Ended'),
+                'grant':grant_id,
+                'text': _('This Grant has ended.'),
+                'subtext': _('Contributions can no longer be made this grant'),
+                'success': False
+            })
+            continue
+
+        if is_grant_team_member(grant, profile):
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant funding blocked'),
+                'grant':grant_id,
+                'text': _('This Grant cannot be funded'),
+                'subtext': _('Grant team members cannot contribute to their own grant.'),
+                'success': False
+            })
+            continue
+
+        if grant.link_to_new_grant:
+            failures.append({
+                'active': 'grant_error',
+                'title': _('Fund - Grant Migrated'),
+                'grant':grant_id,
+                'text': _('This Grant has ended'),
+                'subtext': _('Contributions can no longer be made to this grant. <br> Visit the new grant to contribute.'),
+                'success': False
+            })
+            continue
```

Comment:
^ changes from L130 to L172 can be done in sync ! cause they contain useful checks.
So we'd do all these basic checks and pass the data here